pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -0,0 +1,88 @@
+package component
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/golang/glog"
+
+	adapterCommon "github.com/openshift/odo/pkg/devfile/adapters/common"
+	"github.com/openshift/odo/pkg/devfile/versions/common"
+	"github.com/openshift/odo/pkg/kclient"
+)
+
+// New instantiantes a component adapter
+func New(adapterContext adapterCommon.AdapterContext, client kclient.Client) Adapter {
+	return Adapter{
+		Client:         client,
+		AdapterContext: adapterContext,
+	}
+}
+
+// Adapter is a component adapter implementation for Kubernetes
+type Adapter struct {
+	Client kclient.Client
+	adapterCommon.AdapterContext
+}
+
+// GetContainers iterates through the components in the devfile and returns a slice of the corresponding containers
+func (a *Adapter) GetContainers() []corev1.Container {
```

Comment:
I believe, `GetContainers()` can be made more generic by not binding it to the `Adapter{}` struct and passing the result of `GetAliasedComponents()` as an input paramter.

This way this function becomes a contender to go in the `utils.go` file.

And I feel that `utils.go` would serve purposes across all the implemented `Adapters`.

Reply:
That's a good suggestion, I've moved it to utils.go